src/model/impl/TaskManager.cpp

```diff
@@ -18,7 +18,14 @@ TaskManager::TaskManager(std::unique_ptr<IdGenerator> generator)
 
 ModelResponse TaskManager::Add(const Task& task)
 {
-    TaskId new_id = gen_->GetNextId();
+    TaskId new_id;
+    {
+        std::lock_guard<std::mutex> lock{id_generation_mutex_};
+
+        new_id = gen_->GetNextId();
+    }
+
+    std::lock_guard<std::mutex> lock{modification_mutex_};
```

Comment:
statement above leads to possible confusion between mutexes. People might use wrong mutexes for wrong resources by mistake. I'd prefer single class single mutex, unless necessary.

Reply:
Fixed